src/commands.rs

```diff
@@ -44,7 +44,16 @@ fn represents_several_files(files: &[PathBuf]) -> bool {
 /// Entrypoint of ouch, receives cli options and matches Subcommand to decide what to do
 pub fn run(args: Opts, question_policy: QuestionPolicy) -> crate::Result<()> {
     match args.cmd {
-        Subcommand::Compress { files, output: output_path } => {
+        Subcommand::Compress { mut files, output: output_path } => {
+            // If the output_path file exists and is the same as some of the input files, warn the user and skip those inputs (in order to avoid compression recursion)
+            if output_path.exists() {
+                clean_input_files_if_needed(&mut files, &output_path.canonicalize()?);
```

Comment:
Here you'll need to call `fs::canonicalize()` instead of `Path::canonicalize` because we are using `fs_err` aliased to `fs`, but `Path::canonicalize` will be calling `std::fs::canonicalize` instead.

Reply:
done